runtime/src/serde_snapshot.rs

```diff
@@ -441,6 +441,22 @@ where
             .unwrap_or_else(|err| panic!("Failed to create directory {}: {}", path.display(), err));
     }
 
+    {
+        // inflate 'roots_original'
+        // inserting into roots_original needs to be in order
+        snapshot_prior_roots.sort_unstable();
+        snapshot_prior_roots_with_hash.sort_unstable();
+        let mut roots_original = accounts_db.accounts_index.roots_tracker.write().unwrap();
+        snapshot_prior_roots.into_iter().for_each(|root| {
+            roots_original.roots_original.insert(root);
+        });
+        snapshot_prior_roots_with_hash
```

Comment:
these have to be sorted together so the overall adding is monotonically increasing.

Reply:
I broke this into its own function I can test separately.

Reply:
> these have to be sorted together so the overall adding is monotonically increasing.

and sorry - I wasn't clear. I was commenting I _needed_ to do this and hadn't yet.